peachjam/adapters/adapters.py

```diff
@@ -158,10 +179,78 @@ def update_document(self, url):
             # for stub documents, use the publication document as the source file
             pubdoc = document["publication_document"]
             if pubdoc and pubdoc["url"]:
-                self.download_source_file(pubdoc["url"], doc, title)
+                self.download_source_file(pubdoc["url"], created_doc, title)
         else:
             # the source file is the PDF version
-            self.download_source_file(f"{url}.pdf", doc, title)
+            self.download_source_file(f"{url}.pdf", created_doc, title)
+
+        self.fetch_relationships(document, created_doc)
+        logger.info("Fetching of relationships is complete!")
+
+    def fetch_relationships(self, imported_document, created_document):
+        subject_work = created_document.work
+
+        if imported_document["repeal"]:
+            repealing_work, _ = Work.objects.get_or_create(
+                frbr_uri=FrbrUri.parse(imported_document["repeal"]["repealing_uri"]),
+                title=imported_document["repeal"]["repealing_title"],
+            )
+            self.create_relationship(
+                predicate_idx=1,
+                subject_work=subject_work,
+                object_work=repealing_work,
+                relationship="Repeal",
+            )
+
+        if imported_document["amendments"]:
+            for amendment in imported_document["amendments"]:
+                if amendment["amending_uri"] and amendment["amending_title"]:
+                    amending_work, _ = Work.objects.get_or_create(
+                        frbr_uri=FrbrUri.parse(amendment["amending_uri"]),
+                        title=amendment["amending_title"],
+                    )
+                    self.create_relationship(
+                        predicate_idx=0,
+                        subject_work=subject_work,
+                        object_work=amending_work,
+                        relationship="Amending",
+                    )
+
+        if imported_document["commencements"]:
+            for commencement in imported_document["commencements"]:
+                if (
+                    commencement["commencing_frbr_uri"]
+                    and commencement["commencing_title"]
+                ):
+                    commencing_work, _ = Work.objects.get_or_create(
+                        frbr_uri=FrbrUri.parse(commencement["commencing_frbr_uri"]),
+                        title=commencement["commencing_title"],
+                    )
+                    self.create_relationship(
+                        predicate_idx=2,
+                        subject_work=subject_work,
+                        object_work=commencing_work,
+                        relationship="Commencing",
+                    )
+
+    def create_relationship(
+        self, predicate_idx, subject_work, object_work, relationship
```

Comment:
This method is taking two parameters which identify the relationship: `predicate_ix` and `relationship`, so there is repetition. Also, the `relationship` value is only used in a log.

I suggest dropping the `relationship` parameter and using the `name` value of the predicate object for the log.

Reply:
Nice catch!